src/debuglog.h

```diff
@@ -0,0 +1,52 @@
+/*
+    Copyright (C) 2020 Sebastian J. Wolf and other contributors
+
+    This file is part of Fernschreiber.
+
+    Fernschreiber is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Fernschreiber is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Fernschreiber. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef DEBUGLOG_H
+#define DEBUGLOG_H
+
+
+#include <QObject>
+#include <QQmlEngine>
+
+class DebugLog : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(bool enabled READ isEnabled WRITE setEnabled NOTIFY enabledChanged)
+public:
+
+#ifdef QT_QML_DEBUG
+    DebugLog(QObject* parent = Q_NULLPTR) : QObject(parent), enabled(true) {}
+#else
+    DebugLog(QObject* parent = Q_NULLPTR) : QObject(parent), enabled(false) {}
+#endif
+    static QObject* createSingleton(QQmlEngine* aEngine, QJSEngine* aScript) { return new DebugLog(); }
```

Comment:
Drop these `aEngine`  and `aScript` things, they are unused (and look weird next to everything else)

Reply:
When I removed those, It didn't build, because apparently `qmlRegisterSingletonType` expected them…

Reply:
I mean like this:
```C++
static QObject* createSingleton(QQmlEngine*, QJSEngine*) { return new DebugLog(); }
````

Reply:
Oh, thanks!